File: examples/canonical-iac-pulumi/dotnet/azure/components/identity-component/IdentityComponent.cs
using Pulumi;
using Pulumi.AzureNative.ManagedIdentity;

public sealed class IdentityComponent : ComponentResource
{
    public Output<string> IdentityId { get; private set; }

    public IdentityComponent(string name, string environment, string namePrefix, ComponentResourceOptions? opts = null)
        : base("accb:azure:IdentityComponent", name, opts)
    {
        var stackName = Deployment.Instance.StackName;
        var identity = new UserAssignedIdentity($"{namePrefix}-{environment}-{stackName}-identity", new(), new CustomResourceOptions { Parent = this });
        IdentityId = identity.Id;
        RegisterOutputs(new Dictionary<string, object?> { ["identityId"] = IdentityId });
    }
}

File: examples/canonical-iac-pulumi/python/azure/components/identity_component.py
import pulumi


class IdentityComponent(pulumi.ComponentResource):
    def __init__(self, name: str, environment: str, name_prefix: str, storage_name, eventing_name, secret_ref, opts=None):
        super().__init__("accb:azure:IdentityComponent", name, None, opts)
        stack = pulumi.get_stack()
        self.identity_ref = pulumi.Output.from_input(f"{name_prefix}-{environment}-{stack}-runtime")
        self.register_outputs({"identity_ref": self.identity_ref})
